apiserver/worker/compiler.py

```diff
@@ -269,6 +269,31 @@ def cmd_error_filter(self, cmd_out, cmd_errors):
         return cmd_errors
 
 
+class ReturncodeCompiler(ExternalCompiler):
+    """
+    A compiler that returns an error if the return code is not 0.
+    """
+    def __init__(self, args, separate=False, out_files=[], out_ext=None):
+        ExternalCompiler.__init__(self, args, separate, out_files, out_ext)
+
+    def __str__(self):
+        return "ReturncodeCompiler: %s" % (' '.join(self.args),)
+
+    def cmd_error_filter(self, cmd_out, cmd_errors, returncode):
+        if returncode == 0:
+            return []
+
+        result = [
+            "Error compiling with command {}".format(''.join(self.args)),
+            "Process exited with return code {}".format(returncode),
+            "stdout was:",
+        ]
+        result.extend(line for line in (cmd_out or []) if line is not None)
+        result.append("stderr was:")
+        result.extend(line for line in (cmd_errors or []) if line is not None)
```

Comment:
I know there is a comment elsewhere that it can happen and is protected against many places so it's fine to do so here as well. But do you know how and what causes a line to end up `None` instead of a string?

From what I can see `cmd_out` and `cmd_error` are always the product of a `string.split("\n")` with some items (lines) possibly removed. That shouldn't be adding `None` items.

Reply:
I'm not sure, either. It comes from 5c995e6f2 but there's no indication of why there, either.